modules/solid_properties/src/base/SolidPropertiesApp.C

```diff
@@ -0,0 +1,91 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SolidPropertiesApp.h"
+#include "Moose.h"
+#include "AppFactory.h"
+#include "MooseSyntax.h"
+
+InputParameters
+SolidPropertiesApp::validParams()
+{
+  InputParameters params = MooseApp::validParams();
+  params.set<bool>("use_legacy_material_output") = false;
+  return params;
+}
+
+registerKnownLabel("SolidPropertiesApp");
+
+SolidPropertiesApp::SolidPropertiesApp(InputParameters parameters) : MooseApp(parameters)
+{
+  SolidPropertiesApp::registerAll(_factory, _action_factory, _syntax);
+}
+
+void
+SolidPropertiesApp::registerApps()
+{
+  registerApp(SolidPropertiesApp);
+}
+
+static void
+associateSyntaxInner(Syntax & syntax, ActionFactory & /*action_factory*/)
+{
+  registerSyntaxTask(
+      "AddSolidPropertiesAction", "Modules/SolidProperties/*", "add_solid_properties");
+  registerMooseObjectTask("add_solid_properties", SolidProperties, false);
+  registerMooseObjectTask("add_sp_output", Output, false);
+
+  syntax.addDependency("add_solid_properties", "init_displaced_problem");
+  syntax.addDependency("add_aux_variable", "add_solid_properties");
+  syntax.addDependency("add_variable", "add_solid_properties");
+  syntax.addDependency("add_elemental_field_variable", "add_solid_properties");
+  syntax.addDependency("add_external_aux_variables", "add_solid_properties");
+  syntax.addDependency("add_sp_output", "add_output");
+  syntax.addDependency("add_postprocessor", "add_sp_output");
```

Comment:
It seemed to work fine in the test that covers it?

Reply:
the dependency must happen through another task. But I dont know which one

Reply:
As for the dependencies on `add_soild_properties`, I think those are good already. As for `add_sp_output`, I don't think this actually accomplishes anything yet. I know it was copied from `fluid_properties` module, which has `FluidPropertiesInterrogator` that uses this task. I think `SolidPropertiesInterrogator` might be a useful addition at some point, but I think for now I'll just delete the `add_sp_output` task until it's needed.

Reply:
I would want to see :
```
syntax.addDependency("add_solid_properties", "add_function");
```
not sure why the dependencies are written this way btw.
It should be addDependency(task, pre_req) and I would see adding variables as the prereq to task add_solid_prop not the other way around?

Reply:
I'm not entirely sure what the right thing to do is, but I would say that adding variables is not a prereq for adding solid properties: these UOs should never bring in a variable. I just know that there are a large number of types of MOOSE objects that may need to be constructed with the solid properties UO already existing. Variables were probably chosen here because these same MOOSE objects depend on variables, and the developer didn't want to write all of the different possible dependencies.

Reply:
yes we do have one solid property checked in already that is using getFunction in its constructor, so it depends on Functions.

Reply:
Perhaps we should try getting rid of the `add_solid_properties` task and registering `AddSolidPropertiesAction` to `add_user_object` instead. Then we just change it later if and when we find we need to do something different with solid properties UOs. I'll try this.

Reply:
Ok I think we need `add_solid_properties` because we've added the new MOOSE object syntax `Modules/SolidProperties/*`. What I want to do is just say that `add_solid_properties` has the same dependencies as `add_user_object` - do we know of some way to do that? Seems like
```
syntax.addDependency("add_solid_properties", "add_user_object");
```
wouldn't quite be right because that only gives one direction of the dependence.

Reply:
so UO then solid properties then materials ?

Reply:
@GiudGiud Is this good? If so, can you mark as resolved? LGTM!

Reply:
lgtm